### ISSUE TITLE
[BOOM-2339] A part of the norwegian language pack is missing

### ISSUE THREAD

**Issue by prestonBot** [PrestaShop/PrestaShop#10012]
> This issue has been migrated from the Forge. Read the [original ticket here](http://forge.prestashop.com/browse/BOOM-2339).

- _**Reporter:**_ hwbraathen@gmail.com
- _**Created at:**_ Thu, 2 Feb 2017 06:43:48 +0100

<p>Some files are missing when I select Norwegian language. When I create a new product the "tiny mce" is missing. See attachments.</p>

**Comment by prestonBot** [comment]
> This comment has been migrated from the Forge. Read the [original comment here](http://forge.prestashop.com/browse/BOOM-2339#comment-140444).

- _**Author:**_ guillaumeb
- _**Created at:**_ Fri, 3 Feb 2017 15:15:52 +0100

<p>Hi Helge William Braathen,</p>

<p>This issue was fixed since 1.7.0.4. You'll have to update your shop to fix this problem.</p>

<p>Thanks</p>

**Comment by prestonBot** [comment]
> This comment has been migrated from the Forge. Read the [original comment here](http://forge.prestashop.com/browse/BOOM-2339#comment-140452).

- _**Author:**_ hwbraathen@gmail.com
- _**Created at:**_ Fri, 3 Feb 2017 15:24:18 +0100

<p>sorry Im not mentioned it, I have Version 1.7.0.4 and I still have this problem, only with the Norwegian Language Pack. </p>

<p>In the language file in tiny_mce folder Norwegian have the language location nb-NO, in the PrestaShop DB it is no-NO. Can that be a part of the problem?</p>

**Comment by prestonBot** [comment]
> This comment has been migrated from the Forge. Read the [original comment here](http://forge.prestashop.com/browse/BOOM-2339#comment-140482).

- _**Author:**_ guillaumeb
- _**Created at:**_ Fri, 3 Feb 2017 16:59:17 +0100

<p>qu'en penses-tu ?</p>

**Comment by prestonBot** [comment]
> This comment has been migrated from the Forge. Read the [original comment here](http://forge.prestashop.com/browse/BOOM-2339#comment-140519).

- _**Author:**_ alexeven
- _**Created at:**_ Mon, 6 Feb 2017 09:35:41 +0100

<p>J'allais dire que c'était sûrement lié aux codes langues, comme pour le Slovène. On dirait que tinyMce utilise le bokmal, et nous le nynorsk. Va falloir checker les correspondances et si y'a pas d'autres langues concernées.</p>

**Comment by prestonBot** [comment]
> This comment has been migrated from the Forge. Read the [original comment here](http://forge.prestashop.com/browse/BOOM-2339#comment-150796).

- _**Author:**_ hwbraathen@gmail.com
- _**Created at:**_ Wed, 26 Jul 2017 02:21:55 +0200

<p>In the PrestaShop version 1.7.2.0 this error is ALMOST fixed.</p>

<p>I had to manually change the language code for Norwegian from no to nb (Norwegian bokmål) and then finally the toolbar to showed up.</p>

<p>In Norway we have two written languages.<br/>
In ISO 639-1, nb is the correct language code for norwegian bokmål and nn is the correct language code for norwegian Nynorsk.</p>

<p>Bokmål (nb) is the preferred written standard of Norwegian for 85% to 90% of the population (Wikipedia).</p>

**Comment by prestonBot** [comment]
> This comment has been migrated from the Forge. Read the [original comment here](http://forge.prestashop.com/browse/BOOM-2339#comment-152628).

- _**Author:**_ gustoneo
- _**Created at:**_ Sun, 27 Aug 2017 12:03:30 +0200

<p>Hi,</p>

<p>The same issue when you install Prestashop in French Canadian (Quebec).</p>

**Comment by prestonBot** [comment]
> This comment has been migrated from the Forge. Read the [original comment here](http://forge.prestashop.com/browse/BOOM-2339#comment-152745).

- _**Author:**_ alexeven
- _**Created at:**_ Tue, 29 Aug 2017 14:54:22 +0200

<p><a href="http://forge.prestashop.com/secure/ViewProfile.jspa?name=ssterckx" class="user-hover" rel="ssterckx">STERCKX Sébastien</a> - on avait pas corrigé tous ces soucis de TinyMce ?</p>

**Comment by prestonBot** [comment]
> This comment has been migrated from the Forge. Read the [original comment here](http://forge.prestashop.com/browse/BOOM-2339#comment-157387).

- _**Author:**_ vincentbz
- _**Created at:**_ Fri, 24 Nov 2017 12:04:41 +0100

<p>Ahah encore moi <a href="http://forge.prestashop.com/secure/ViewProfile.jspa?name=Louise+Bonnard" class="user-hover" rel="Louise Bonnard">Louise Bonnard</a> !</p>

<p>T'aurais moyen de jeter un oeil stp ? ^^</p>

**Comment by prestonBot** [comment]
> This comment has been migrated from the Forge. Read the [original comment here](http://forge.prestashop.com/browse/BOOM-2339#comment-172788).

- _**Author:**_ marion_francois
- _**Created at:**_ Fri, 27 Jul 2018 13:47:56 +0200

<p>Hello <a href="http://forge.prestashop.com/secure/ViewProfile.jspa?name=khouloud.belguith" class="user-hover" rel="khouloud.belguith">Khouloud BELGUITH</a></p>

<p>Can you try to reproduce on the last version please ?</p>

<p>Thanks <img class="emoticon" src="http://forge.prestashop.com/images/icons/emoticons/smile.png" height="16" width="16" align="absmiddle" alt="" border="0"/></p>

**Comment by prestonBot** [comment]
> This comment has been migrated from the Forge. Read the [original comment here](http://forge.prestashop.com/browse/BOOM-2339#comment-172809).

- _**Author:**_ khouloud.belguith
- _**Created at:**_ Fri, 27 Jul 2018 15:04:54 +0200

<p>Hi <a href="http://forge.prestashop.com/secure/ViewProfile.jspa?name=marion_francois" class="user-hover" rel="marion_francois">Marion F</a>,</p>

<p>Yes, I manage to reproduce the issue with PS1.7.4.2.<br/>
 <span class="image-wrap" style=""><a id="69930_thumb" href="http://forge.prestashop.com/secure/attachment/69930/69930_screenshot-1.png" title="screenshot-1.png" file-preview-type="image" file-preview-id="69930" file-preview-title="screenshot-1.png"><img src="http://forge.prestashop.com/secure/thumbnail/69930/_thumb_69930.png" style="border: 0px solid black" /></a></span> </p>

<p>Best regards, Khouloud</p>

**Comment by leos2000171** [comment]
It also happens when installing Prestashop (1.7.6.4) with Trad. Chinese. It also missing tiny_mce js file.

But I found the solution temporary. Which is going to 
https://www.tiny.cloud/get-tiny/language-packages/
and download the language js file.
However, when download Trad. Chiese, be aware of the file name.
zh_TW.js <-- The file download from above site
tw.js <-- The correct file name in Prestashop

Once downloaded the file, put that file into "/js/tiny_mce/langs" folder.
That should be found.

**Comment by boo-code** [comment]
This is the same defect as #13131 and is covered by #42335.

The back office hands the employee's iso code to TinyMCE unchecked, and TinyMCE breaks when there is no language file for it. Norwegian is one of eight shipped iso codes with no matching file:

```
shipped BO languages with NO TinyMCE language file:  hi mk mx no qc sq tw vn
```

which is why the editor does not come up when the employee language is Norwegian, exactly as reported here in 2017.

TinyMCE does carry Norwegian, under `nb.js` rather than `no.js`, so the fix maps it rather than falling back to English:

```
no -> nb
```

Same for `mx` to `es_MX`, `tw` to `zh_TW`, `vn` to `vi`, and `qc` to `fr` since there is no `fr_CA`. Only `hi`, `mk` and `sq` have nothing at all and still get English.

Worth closing this one as a duplicate of #13131 once that lands.
